Docstring edits

jrfiedler opened the pull request:
<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->
<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

Issue #203

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. -->
<!-- You can skip this if you're fixing a typo or adding an app to the Showcase.  -->
<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->


<!-- In addition to that please answer these questions: -->

**What kind of change does this PR introduce?**

Mostly docstring edts.

In a few places I've added info to docstrings, and those should probably get some extra checking to make sure they're correct.

In one place I've changed a parameter name (see next question).

**Does this PR introduce a breaking change?**

This currently changes one function parameter name. This is probably the most controversial change here. I'm happy to change it back if the old name is preferred.

In `abstract_model.py`, in the method `TabModel.save_model()`, I've changed `path` to `filepath`. I made this change because the following method, `TabModel.load_model()` uses `filepath`, and there are common submodules in the standard library called `path` (`os.path`, `sys.path`). Also, I didn't see any code that references the parameter name in .py or .ipynb files. But, again, I'm happy to change this back if the old name is preferred.

**What needs to be documented once your changes are merged?**

Nothing. The changes here are mostly in the documentation.

**Closing issues**

closes #203

eduardocarvp commented:
Thanks, @jrfiedler ! 
I had a quick look on the changes and it should be good. I just need to test it along with the Sphinx document generation and I think we can merge.

The variable name change seems ok, I personally prefer `filepath` as well.

Optimox commented on pytorch_tabnet/utils.py:
```diff
@@ -58,24 +58,32 @@ def create_dataloaders(
+        set to True to drop the last incomplete batch, if the dataset size is not
+        divisible by the batch size. If False and the size of dataset is not
+        divisible by the batch size, then the last batch will be smaller
+
```
Could you add `pin_memory` please?
It's a boolean value that decides whether to pin gpu memory during training.

Optimox reviewed (changes requested):
Thanks!
Did you pass through all the code?

jrfiedler commented:
@Optimox, I've reverted the parameter name and I added `pin_memory` to that docstring.

And yes, I have looked through all of the Python files for issues like these, and I fixed all that I found.

Optimox commented:
@jrfiedler could you squash all your commits into one please? Then it's good to go!

Optimox commented:
Thank you very much @jrfiedler !
